### PR TITLE
Fix prototype pollution vulnerabilities, fix failing tests

### PR DESCRIPTION
**Pull request by joeattardi**

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -14923,7 +14963,7 @@
     });
 
     test('should throw a TypeError when receiving a non-array `array` argument', 1, function() {
-      raises(function() { _.spread(4, 2); }, TypeError);
```
Comment (jdalton) [inline test/test.js]: Does `raises` still work? In v3 we avoided `throws` because of some WebKit engines (older for sure by now).

Comment (joeattardi) [reply test/test.js]: When I ran the tests it was throwing errors that `raises` was not defined. Was not aware of the issues with `throws` though.

Comment (jdalton) [reply test/test.js]: That's okay, no worries. Thank you!

**Comment by jbegleiter-sf** [comment]
@jdalton is this released?

**Comment by remarkablemark** [comment]
This is not, I created my own repository for npm package with the fix: https://github.com/remarkablemark/lodash

**Comment by jbegleiter-sf** [comment]
Thanks @remarkablemark , @jdalton any chance we could have this released as part of official? Would save me a headache.

**Pull request merged**